### ISSUE TITLE
App crashing on Duo 2

### ISSUE THREAD

**Issue by Raxcental** [DenyTheFlowerpot/Scrunch#2]
App does not work as intended on Microsoft Surface Duo 2. App does not play sound when opening or closing the device. The app also crashes repeatedly. While this issue is known, I am posting it here anyway for public record and documentation.

**Comment by DenyTheFlowerpot** [comment]
Thank you! SD2 support will have to wait for a while, I currently only have time to prepare an upcoming release with a fix for SD1 support as well as quiet time scheduling.

**Comment by Raxcental** [comment]
Any news on updates or plans?

**Comment by DenyTheFlowerpot** [comment]
Sorry, I am extremely busy IRL currently. I'll try to get back on track some time this month.

**Comment by Raxcental** [comment]
Is this still in the works?

**Comment by Raxcental** [comment]
Just checking in again. Are you still planning on continuing this project?

**Comment by DenyTheFlowerpot** [comment]
Sorry for the delay. Are you still interested in Scrunch? I would need logcat logs to work on SD2 support.

**Comment by Raxcental** [comment]
Yes. I'm still interested in scrunch. I'd be willing to provide logs. I just need instructions on how to find them

**Comment by DenyTheFlowerpot** [comment]
Are you familiar with logcat? I would need some logcat logs taken while you're opening and closing the device. Currently Scrunch relies 100% on parsing logcat, so I need to know the format of the system messages sent when it is open/closed.

**Comment by Raxcental** [comment]
I'm not familiar but I'm certain I can look it up and manage to get it figured out. It'll have to wait until after work though. I'll get them to you as soon as I can.

**Comment by DenyTheFlowerpot** [comment]
Thanks. Take your time, I sure took mine after all.

**Comment by DenyTheFlowerpot** [comment]
Thanks! Got what I needed, here's a test build. Does it work for you?
In the meantime, I will delete your message with the log just in case there's personal info on there.

[app-release.apk.zip](https://github.com/DenyTheFlowerpot/Scrunch/files/9648458/app-release.apk.zip)

**Comment by Raxcental** [comment]
Okay. So. Good news is that the app no longer crashes. 
Bad news is that the selected sounds do not play. 

I made sure the service switch is toggled on. I have the persistent notification in my notification shade. I have granted the permission via ADB. The sound just isn't playing :(

**Comment by DenyTheFlowerpot** [comment]
What format are the sounds?

**Comment by Raxcental** [comment]
They are OGG files. The sound plays when adjusting the volume inside scrunch. It just doesn't play when opening or closing the device.

**Comment by DenyTheFlowerpot** [comment]
Mind taking another log for me? I'm testing locally with the previous one and the parsing I added for SD2 seems to work, so there might be something else going on.

**Comment by Raxcental** [comment]
I made sure to open and close the device a few times before grabbing this one. Lets hope if it sheds some light.
[LogcatLog.zip](https://github.com/DenyTheFlowerpot/Scrunch/files/9649726/LogcatLog.zip)

**Comment by DenyTheFlowerpot** [comment]
Not seeing any movement from Scrunch in this one. I have a hunch as to why. If you are so kind, I would need 2 more things:
1. Another log, but this time please force close scrunch first, then start the log, then open Scrunch. 
2. When you open adb shell, what prompt do you see? I mean what text is shown before the $ sign.

Thank you for your assistance with this!

**Comment by Raxcental** [comment]
When obtaining the logcat files there is no $ because I'm running ADB commands through the windows terminal. I did however grant the initial ADB permission (I think properly) via web ADB. Though it didn't give me a message after entering the grant permission command.

Heres the latest log after following you directions.
[LogcatLog.zip](https://github.com/DenyTheFlowerpot/Scrunch/files/9649798/LogcatLog.zip)

**Comment by DenyTheFlowerpot** [comment]
Thanks. The $ prompt would have been important as Scrunch uses it to detect the device it is running on. But let's exclude that for a moment and if nothing else works, maybe we'll get back to it.

Reading through the logs, I discovered that the reason I'm not seeing anything useful is that in release mode, debug logs are not printed. This is a good thing to have for commercial apps, but not so much for hobby ones, especially when you're trying to debug an issue. Sorry! 

I've made a debug build, if you still feel like pulling logs. You need to uninstall the normal one first, otherwise Android will complain. Perhaps it would be good to start the log and then install the debug build.
If you don't feel like going through this again I totally understand it and appreciate your help thus far! It has been very valuable, as Microsoft isn't even selling Surface Duos in my country.

[app-debug.apk.zip](https://github.com/DenyTheFlowerpot/Scrunch/files/9649881/app-debug.apk.zip)
